### ISSUE TITLE
Ubuntu Docker Permissions Issue

### ISSUE THREAD

**Issue by ntorba** [aave/lens-protocol#56]
Hi Lens team, 

I tried to follow the [quick setup docs](https://docs.lens.dev/docs/quick-setup), but I am running into a docker permissions issue (likely because I am running this on ubuntu). I wanted to post this here in case your team has run into this before, otherwise, I'll see if I can figure it out myself. 

When I run `sudo docker-compose up`, I see this output: 
```bash
...
Successfully built bcf4f82a324a
Successfully tagged lens-protocol_contracts-env:latest
Starting lens-protocol_contracts-env_1 ... done
Attaching to lens-protocol_contracts-env_1
contracts-env_1  | mv: cannot create directory '/src/node_modules': Permission denied
```
The permission denied line is troublesome, but I can still exec into the container. 

I exec into the container and run `npm run compile, I get this output: 
```bash
I have no name!@ac2da960baa4:/src$ npm run compile

> lens-protocol@1.0.0 compile
> SKIP_LOAD=true hardhat clean && SKIP_LOAD=true hardhat compile

An unexpected error occurred:

Error: EACCES: permission denied, mkdir '/.config'
    at Object.mkdirSync (node:fs:1336:3)
    at mkdirsSync (/node_modules/fs-extra/lib/mkdirs/mkdirs-sync.js:31:9)
    at Object.mkdirsSync (/node_modules/fs-extra/lib/mkdirs/mkdirs-sync.js:36:14)
    at getConfigDirSync (/node_modules/hardhat/src/internal/util/global-dir.ts:21:6)
    at Object.hasConsentedTelemetry (/node_modules/hardhat/src/internal/util/global-dir.ts:103:21)
    at main (/node_modules/hardhat/src/internal/cli/cli.ts:136:49) {
  errno: -13,
  syscall: 'mkdir',
  code: 'EACCES',
  path: '/.config'
}
```

It looks docker will not allow npm to create the directories it needs to successfully complete the commands. 

Please let me know if your team knows a fix to this! 

Thanks, 

Nick

**Comment by ltzMaxwell** [comment]
RUN npm install -g --unsafe-perm=true --allow-root

This helps me.

**Comment by ntorba** [comment]
I think I made a step forward. I realized the  `I have no name!` section at the terminal meant that I wasn't logged in as the `node` user. 

I fixed this by running `sudo docker-compose exec --user node contracts-env bash` (emphasis on the `--user node` arg).

I now get a new error when running `npm run compile`:

```bash
node@d0219ab36c84:/src$ npm run compile

> lens-protocol@1.0.0 compile
> SKIP_LOAD=true hardhat clean && SKIP_LOAD=true hardhat compile

Error HH502: Couldn't download compiler versions list. Please check your connection.

For more info go to https://hardhat.org/HH502 or run Hardhat with --show-stack-traces
```

I saw [this answer](https://github.com/NomicFoundation/hardhat/issues/1246#issuecomment-1054224554) say to use a proxy, but I'm not sure how that applies to my case?
